src/stream_base.h

```diff
@@ -160,7 +160,8 @@ class StreamBase : public StreamResource {
  public:
   enum Flags {
     kFlagNone = 0x0,
-    kFlagHasWritev = 0x1
+    kFlagHasWritev = 0x1,
+    kFlagNoShutdown = 0x2
```

Comment:
You're probably doing this so you don't have to update too much code but shouldn't it be opt-in with `kFlagHasShutdown`?